services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/assign_a_diagnostic.tsx

```diff
@@ -110,9 +162,70 @@ const FilterTab = ({ activeFilter, filter, setFilter, number, }) => {
 const AssignADiagnostic = ({ history, assignedPreTests, }) => {
   document.title = getPageTitle("Assign", "Diagnostic")
   const [filter, setFilter] = React.useState(ALL)
+  const [classCount, setClassCount] = React.useState<number>(null)
+  const [recommendationsCount, setRecommendationsCount] = React.useState<number>(null)
+  const [recommendedDiagnosticsByLevel, setRecommendedDiagnosticsByLevel] = React.useState<RecommendedDiagnosticsByLevelInterface>({})
+
+  React.useEffect(() => {
+    getCompleteHistory()
+  }, [])
+
+  function getCompleteHistory() {
+    requestGet('/diagnostic_insights/complete_history', (body) => {
+      let classCount = 0
+      let recommendationsCount = 0
+      const newRecommendedDiagnosticsByLevel = {
+        [STARTER]: [],
+        [INTERMEDIATE]: [],
+        [ADVANCED]: [],
+        [ELL_STARTER]: [],
+        [ELL_INTERMEDIATE]: [],
+        [ELL_ADVANCED]: []
+      }
```

Comment:
Do we have an interface defined for this data shape?  Knowing what shape the front-end ultimately wants the data to end up in would be useful when we refactor this endpoint into something specific for this purpose.  We can just talk about it at that time, but I figured I'd flag it now so it's on our minds.

Reply:
@anathomical we do have an interface, defined above at the state-level, but I'm referencing it here now to be clearer. However, I don't think we want to change the shape of the data the API is returning to match this, because it ultimately needs to get passed through our frontend `generateRecommendations` function.